src/quo2/components/list_items/channel.cljs

```diff
@@ -45,30 +42,25 @@
          :emoji-background-color (colors/theme-alpha channel-color 0.1 0.1)
          :emoji                  emoji}]
        [quo2.text/text
-        {:style  (merge {:margin-left 12}
-                        (when (and (not locked?) muted?)
-                          {:color (if (theme/dark?) colors/neutral-60 colors/neutral-40)}))
+        {:style  (cond-> {:margin-left 12}
+                   (and (not locked?) muted?)
+                   (assoc :color (colors/theme-colors colors/neutral-40 colors/neutral-60)))
          :weight :medium
-         :size   :paragraph-1} (str "# " name-text)]]
-      [rn/view
-       {:style {:height          20
-                :justify-content :center}}
-       (when (and (not locked?)
-                  muted?)
-         [quo2.icons/icon :i/muted
-          {:size            20
-           :color           colors/neutral-40
-           :container-style {:margin-right 1
-                             :margin-top   2}}])
-       (when (and (not locked?)
-                  (not muted?)
-                  (pos? (int mentions-count)))
-         [rn/view
-          {:style {:margin-right 2
-                   :margin-top   2}}
-          [quo2.counter/counter {:override-bg-color channel-color} mentions-count]])
-       (when (and (not locked?)
-                  (not muted?)
-                  (not (pos? (int mentions-count)))
-                  unread-messages?)
-         [unread-grey-dot :unviewed-messages-public])]]]))
+         :size   :paragraph-1}
+        (str "# " name-text)]]
+      (when-not locked?
+        [rn/view {:style {:height 20 :justify-content :center}}
+         (cond
+           muted?
+           [quo2.icons/icon :i/muted
+            {:size            20
+             :color           colors/neutral-40
+             :container-style {:margin-right 1 :margin-top 2}}]
+
+           (pos? (int mentions-count))
+           [rn/view {:style {:margin-right 2 :margin-top 2}}
+            [quo2.counter/counter {:override-bg-color channel-color}
+             mentions-count]]
+
+           unread-messages?
+           [unread-grey-dot :unviewed-messages-public])])]]))
```

Comment:
Refactor, we had:
 
 ```clojure
 [rn/view 
 (when a
   ,,,)
 (when (and (not a) b)
   ,,,)
 (when (and (not a) (not b) c)
   ,,,)
 ]
 ```
 and now:
 
 ```clojure
 [rn/view
 (cond
   a ,,,
   b ,,,
   c ,,,)
 ]
 ```

Reply:
I extracted the `locked?` condition because if it's true, in the previous code, we would be returning:

`[rn/view {,,,} nil]`

In this case, the UI looks the same if we just return `nil`. Do you still think I should add that `not locked?` for each one?

And talking about `when`. I use structural editing all the time, and I haven't faced problems manipulating `cond`'s pairs.🤔

I added this refactoring because if you read the new `cond` code, since conditions are simpler, it's easier (IMO) to understand what we are checking to return a specific hiccup.

Anyway, I'm always open to revert these kinds of changes since it's a matter of tastes, so if you find it worse, please let me know 👍

Reply:
I think the refactor makes sense @ulisesmac. My point was that even when some duplication can be removed (as you did), I sometimes lean more on the side of convenience to manipulate the s-exps. Maybe my tools are rusty?

> And talking about when. I use structural editing all the time, and I haven't faced problems manipulating cond's pairs.

There's no pair in the `cond` that I can manipulate as a single expression in Clojure, so I don't understand how you manipulate the "pair". In CL `cond` looks like this:

```lisp
(cond ((> 3 4) "variant 1")
      ((> 4 2) "variant 2")
      (t "always valid variant"))
```

The same in Clojure would look like:

```clojure
(cond (> 3 4) "variant 1"
      (> 4 2) "variant 2"
      :else "always valid variant")
```

So Clojure uses less parentheses. But this slight "simplification" affects me on a daily basis, for example:

- I can't easily copy/cut the pair *test* AND *result*.
- I can't fold the pair *test* AND *result* because there's no s-exp to fold, I either fold the *test* OR the *result*.
- I can't comment/uncomment the *test* AND *result*.
- The *test* and *result* are not inside a list, so I can't for example *raise* it, etc.

> Anyway, I'm always open to revert these kinds of changes since it's a matter of tastes, so if you find it worse, please let me know

I never suggest reverting things that I think are personal preference. To me, if it's personal preference and it's aligned with what the rest of the codebase does there's nothing wrong with the person's choice. That's why I started saying it's a nitpick.

Please, keep the code @ulisesmac :)

Reply:
@ilmotta I understand your problems manipulating the code, I use intelliJ + Cursive and if I want to do something with the pair I just need to select the pair, not as straightforward as if it were an s-exp, but not bad.

I recorded a small video manipulating this cond, I don't find it hard to manipulate.

[Screencast from 2023-07-04 15-37-42.webm](https://github.com/status-im/status-mobile/assets/90291778/03273bf1-4edf-499c-954c-76c84aeca2cb)

Reply:
Thank you @ulisesmac, it's definitely easier on your side. I see you expand the region to get the two expressions (test and result).

@yqrashawn, IIRC you use Lispy in Emacs. How do you select or manipulate in general pairs of expressions that are not surrounded by parentheses? For example the key + value of a map, or the test + result of a `cond`, or the `let` binding name + value, etc.